src/main/java/com/marklogic/appdeployer/command/hosts/AssignHostsToGroupsCommand.java

```diff
@@ -0,0 +1,60 @@
+	
+    public AssignHostsToGroupsCommand() {
+        setExecuteSortOrder(SortOrderConstants.ASSIGN_HOSTS_TO_GROUPS);
+        setUndoSortOrder(SortOrderConstants.UNASSIGN_HOSTS_FROM_GROUPS);
+    }
+
+	@Override
+	public void execute(CommandContext context) {
+		Map<String, String> hostGroups = context.getAppConfig().getHostGroups();
+	    AdminManager adminManager = context.getAdminManager();
+		HostManager mgr = new HostManager(context.getManageClient());
+		for (Map.Entry<String, String> entry : hostGroups.entrySet()) {
+			String hostName = entry.getKey();
+			String groupName = entry.getValue();
+			// First, ensure a Manage server exists for the group
+			ServerManager serverMgr = new ServerManager(context.getManageClient(), groupName);
+			try {
+				String serverResponse = serverMgr.getPropertiesAsJson(MANAGE_SERVER_NAME, "group-id="+groupName);
+				System.out.println("serverResponse: " + serverResponse);
+	        } catch (Exception e) {
+	        	if (e.getMessage().contains("404 Not Found")) {
+	                String manageServerPayload = format(MANAGE_SERVER_JSON_TEMPLATE, MANAGE_SERVER_NAME, groupName);
+	        		serverMgr.save(manageServerPayload);
+	        	} else {
+	        		logger.error("Error checking for Manage server in target group: " + e.getMessage());
+	        		throw e;
+	        	}
+	        }
+			mgr.setHostToGroup(hostName, groupName);
```

Comment:
Couple things here - AdminManager has an invokeActionRequiringRestart method that you should use here. The benefit is that it gives you a chance to determine whether or not a restart needs to happen - i.e. if the hosts are already setup, then I don't think a restart would happen, right? 

Two - I ran into an error locally, because the call to waitForRestart hits port 8001. In the test I ran with a 3-host cluster, I moved host1 to the other group, which caused the call to 8001 to fail. So I think you have to create both a Manage server and an Admin server in the new group?